### PR TITLE
Add subfeatures details

### PR DESCRIPTION
**Pull request by Akusem**
I have finally some time available to work on the issue #1234, this is basically the work I had done last time we had discussed about, already 2 weeks ago. 
It display subfeatures (like transcript, exon or CDS) information. To avoid slowing down the browser, only the transcripts and mRNA are displayed until the user click on `load additional subfeatures` .

Ex before clicking:
image

After:
image

### CODE REVIEW

**Comment by cmdcolin** [comment]
@Akusem thanks for submitting this

I tested it out and it's pretty good, especially as a first pass. It might be that it is acceptable as is then

Some random comments

1) I might not use by default `const subfeaturesOmit = ['gene_id', 'gene_name', 'gene_type', 'gene_status']`. Even this is duplicated across multiple transcript subfeatures it is maybe not the best default setting to have as people might want to see these fields
2) The subfeatures are technically probably children of a specific transcript but they are loaded below all the transcripts
screenshot from wormbase with the JBrowse 1 featuresequence plugin
3) I feel like, probably a really awesome ideal would be to de-emphasize subfeatures like cds, exon (which you have already done) and have something like this


wormbase org_tools_genome_jbrowse-simple__data=data%2Fc_elegans_PRJNA13758 loc=II%3A159133 160002 tracks=Curated_Genes highlight=

That is a screenshot of the featuresequence plugin in jbrowse 1 on wormbase

Of course that is a longer PR so we might be able to accept this as is, but I want to open our mind to the future of what a transcript feature detail panel *could be*

**Comment by cmdcolin** [comment]
Random other comment: non gene features have an "empty subfeatures area" that is visible below their details

localhost_3000__config=test_data%2Fvolvox%2Fconfig json

**Comment by Akusem** [comment]
Hi @cmdcolin,
I've fixed the 'empty subfeatures area', removed `subfeaturesOmit`, and finally i've modified the omited field to allow subfeatures to display `start`, `end` and `name` field, who can be useful to the user I think ?

I have to say that the output of featuresequence plugin in jbrowse 1 look superbe, and I agree that it would be much nicer to have than a simple list of subfeatures like now ! 

Just two remark:
1. First, I don't think the `feature detail` panel is large enought for this without looking too compressed, so we will maybe have to enlarge it ?
2. Secondly, I'm not sure on how to access the sequence to display from BaseFeaturesDetails ? Maybe from `model` props ?

File: packages/core/BaseFeatureWidget/BaseFeatureDetail.tsx [hunk 1]
```diff
@@ -223,14 +223,107 @@ export const Attributes: FunctionComponent<AttributeProps> = props => {
...
+// Accompanied with a button to load additionnal subfeature like exon/cds.
+export const BaseSubFeatures = (props: BaseProps) => {
+  const { feature, descriptions } = props
+  const [subfeaturesLoaded, setSubfeaturesLoaded] = useState(false)
+  let subfeaturesToRender = getSubfeaturesToRender(feature.subfeatures, [])
+  // If not loaded, display only transcript and mRNA
+  if (!subfeaturesLoaded) {
+    subfeaturesToRender = subfeaturesToRender.filter(subfeature =>
+      ['transcript', 'mRNA'].includes(subfeature.title),
+    )
```
Comment (rbuels) [inline packages/core/BaseFeatureWidget/BaseFeatureDetail.tsx:285]: This filtering assumes a specific structure of subfeatures, which is not appropriate for this completely-generic feature detail implementation. Could you remove it?

**Review by rbuels — COMMENTED** [review]
sorry for the slow response @Akusem. we're still interested in merging this, I think it looks quite good in general. My main concern is that it is specifically for gene features (assumes transcript and mRNA feature types), but this feature detail implementation is meant to be very general and can't assume any specific structure of subfeatures.

File: packages/core/BaseFeatureWidget/BaseFeatureDetail.tsx (hunk 1 above)
Comment (Akusem) [reply packages/core/BaseFeatureWidget/BaseFeatureDetail.tsx:285]: @rbuels If I understand correctly it should only display a `load subfeatures` button at start and then load every subfeatures ?
Example:
image

image

Comment (rbuels) [reply packages/core/BaseFeatureWidget/BaseFeatureDetail.tsx]: What I meant is, this is specific to "transcript" and "mRNA" subfeatures. This feature track is used to display many things, not just genes. Filtering here for only mRNA and transcript subfeatures means that other types of features won't be able to display their subfeatures. I see you removed it in 9588d98 anyway, so you probably understand what I mean already. :-)

**Comment by cmdcolin** [comment]
@rbuels I made an alternative to this PR in https://github.com/GMOD/jbrowse-components/pull/1529

It is a little more straightforward to me, and includes nested subfeatures

**Comment by cmdcolin** [comment]
We could probably merge this as is and then follow up with #1529

**Comment by cmdcolin** [comment]
Thanks a bunch @Akusem for pushing on this issue, it was really great :)

**Pull request merged**